packages/odyssey-react/src/utils/useSharedRef.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { ForwardedRef, RefObject, useEffect, useRef } from "react";
+
+export function useSharedRef<T>(ref?: ForwardedRef<T>): RefObject<T> {
```

Comment:
awesome nice use of generics here

Reply:
I think `ref` should/could stay optional. Someone could use this hook inside another hook, like a `useFocus` hook that takes a `ref` as an optional param. Since `ForwardedRef` can be `null`. There should be no impact on behavior if the ref is `null` or `undefined` value or just `undefined`?